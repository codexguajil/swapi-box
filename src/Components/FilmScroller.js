import React from 'react';
import propTypes from 'prop-types';

export const FilmScroller = ({films}) => {

    let randomNum = Math.floor(Math.random() * films.length)

   return (
       <div id="film">
            <div id="crawl">
            <h1>welcome</h1>
            {films.length > 0 &&
                    <h3>
                        {films[randomNum].opening_crawl}
                        <p>{films[randomNum].title}</p>
                        <p>{films[randomNum].release_date}</p>
                    </h3>
            }
        </div>
       </div>
   )
}

FilmScroller.propTypes = {
    films: propTypes.array.isRequired
}