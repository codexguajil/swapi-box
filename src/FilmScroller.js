import React from 'react';

export const FilmScroller = ({films}) => {

    let randomNum = Math.floor(Math.random() * films.length)

    console.log(films)

   return (
       <div>
           <h1>welcome</h1>
           {films.length > 0 &&
                <h3>
                    {films[randomNum].opening_crawl}
                </h3>
           }
       </div>

   )

}

