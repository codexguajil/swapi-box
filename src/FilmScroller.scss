body {
    background-color: hsl(0, 0%, 0%);
    overflow: hidden;
}

#film {
    // background-color: hsl(0, 0%, 0%);
    color: rgb(216, 216, 22);
    display: flex;
    font-family: 'Pathway Gothic One', sans-serif;
    font-size: 3rem;
    font-weight: 600px;
    height: 800px;
    justify-content: center;
    letter-spacing: 6px;
    line-height: 150%;
    perspective: 400px;
    top: -100px;
    width: 20%;
}

#crawl {
    animation: crawl 60s linear;
    // background-color: hsl(0, 0%, 0%);
    position: relative;
    top: -100px;
    transform-origin: 50% 100%;
}

@keyframes crawl {
    0% {top: 0%;
        transform: rotateX(20deg) translateZ(0)
    }
    100% {top: -6000px;
          transform: rotateX(25deg) translateZ(-2500px);
    }
}