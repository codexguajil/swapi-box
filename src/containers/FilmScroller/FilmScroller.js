import React, { Component } from 'react';
import { connect } from 'react-redux';

class FilmScroller extends Component {
  
  renderScroll = () => {
    return this.props.movies.map(movie => {
      return (
       <h1 key={movie.episode_id}>{movie.title}</h1>
      )
    })
  }

   render() {
     console.log(this.props)
     return (
     <div className="film">
      {this.renderScroll()}     
     </div>
     )
   } 
}

export const mapStateToProps = (state) => ({
  movies: state.movies,
})

export default connect(mapStateToProps)(FilmScroller);