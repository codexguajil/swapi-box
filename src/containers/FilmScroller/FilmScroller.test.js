import React from 'react';
import FilmScroller from './FilmScroller';
import { shallow } from 'enzyme';

describe ("FilmScroller", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(
          <FilmScroller />
        )

        expect(wrapper).toMatchSnapshot()
    })
})