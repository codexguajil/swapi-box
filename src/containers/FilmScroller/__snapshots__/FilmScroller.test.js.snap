// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmScroller should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
